### ISSUE TITLE
`list.copy` should return a contravariant value

### ISSUE THREAD

**Issue by KotlinIsland** [KotlinIsland/basedmypy#535]
```py
a = [1, 2, 3]
b: list[object] = a.copy()  # error: invariance, expected no error
```
